File: fsharp/secret-handshake/SecretHandshake.fs
module SecretHandshake

let commands number = 
    let set = [(1,"wink");(2,"double blink");(4,"close your eyes");(8,"jump")]
    let deciphered = set |> List.filter (fun x -> fst x &&& number = fst x) |> List.map snd
    if (16 &&& number = 16) then List.rev deciphered else deciphered

File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Linq;

public static class SecretHandshake
{
    private static readonly (int code, string text)[] commands = new[]
    { (1, "wink"), (2, "double blink"), 
      (4, "close your eyes"), (8, "jump") };

    private const int reverseCode = 16;

    public static string[] Commands(int commandValue)
    {
        var result = commands
            .Where(c => (c.code & commandValue) == c.code)
            .Select(c => c.text);

        return (reverseCode & commandValue) == reverseCode 
            ? result.Reverse().ToArray() 
            : result.ToArray();
    }
}
